### PR TITLE
[T3A3][T09-A1]Lim Shun Xian

### PR DESCRIPTION
**Pull request by Crsytral**
Split Address into more classes

### CODE REVIEW

**Comment by Crsytral** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -9,10 +9,19 @@
...
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be entered in this format: a/BLOCK, STREET, UNIT, POSTAL_CODE";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
+    public static final int INDEX_OF_BLOCK = 0;
+    public static final int INDEX_OF_STREET = 1;
+    public static final int INDEX_OF_UNIT = 2;
+    public static final int INDEX_OF_POSTALCODE = 3;
+    
     public final String value;
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Address.java:20]: What is the point of 'value' variable now that you have separate classes to maintain the components in the address?

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -56,4 +70,81 @@ public int hashCode() {
     public boolean isPrivate() {
         return isPrivate;
     }
+    
+}
+
+/**
+ * Represents the Block component of a Person's address in the address book.
+ */
+class Block{
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Address.java:79]: If a class is not a nested class, it is always good to have a separate file for the class. In this case nested class are ok but your implementation is not a nested implementation.

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -56,4 +70,81 @@ public int hashCode() {
...
+	private String block;
+	
+	public Block(String block){
+		this.block = block;
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Address.java:83]: indentation issue??

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -56,4 +70,81 @@ public int hashCode() {
...
+		this.postalCode = postalCode;
+	}
+	
+	public String getPostalCode() {
+		return postalCode;
+	}
+
+	public void setPostalCode(String postalCode) {
+		this.postalCode = postalCode;
+	}
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Address.java:149]: You have not updated the tests or the documents to reflect the changes. These needs to be updated as well.

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -26,7 +35,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        String[] addressDetails = trimmedAddress.split(", ");
+        block = new Block(addressDetails[INDEX_OF_BLOCK]);
+        street = new Street (addressDetails[INDEX_OF_STREET]);
+        unit = new Unit (addressDetails[INDEX_OF_UNIT]);	
+        postalCode = new PostalCode (addressDetails[INDEX_OF_POSTALCODE]);
+        this.value = block.getBlock()+", "+street.getStreet()+", "+unit.getUnit()+", "+postalCode.getPostalCode();
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Address.java:43]: consider doing this in the toString() method. If not there can be duplicate inconsistent data right?

**Review by nirandiw — COMMENTED** [review]
@Crsytral Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**